scripts/deploy

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+# Pass in app as either staging or production first -- must match what you seen in git remote -v
+# Examples:
+# ./deploy staging
+# ./deploy staging --force
+# ./deploy production
+
+app=$1
+
+shift
+
+sha=`git rev-parse HEAD 2>/dev/null`
+current_branch=`git symbolic-ref HEAD 2> /dev/null | sed -e 's/refs\/heads\///'`
+
+# echo "current_branch is $current_branch"
+# echo "sha is $sha"
+# echo "app is $app"
+
+if [ "$app" == "" ]; then
+  echo "You must specify a param value for the app, such as production or staging"
+  exit 1
+fi
+
+if [[ "$app" == "production" && "$current_branch" != "master" ]]; then
+  echo "You can only deploy master to production!"
+  exit 1
+fi
+
+echo Running:
+echo git push $app $current_branch:master $@
+git push $app $current_branch:master $@
+
+echo Setting heroku config DEPLOYMENT_SHA
+echo heroku config:set DEPLOYMENT_SHA=$sha --remote $app
+heroku config:set DEPLOYMENT_SHA=$sha --remote $app
+
+# Run migrations always just in case
+heroku run rake db:migrate --remote $app
+heroku restart --remote $app
```

Comment:
@dylangrafmyre @justin808 What about instead of the `scripts/deploy` approach, we capture the `SOURCE_VERSION` environment variable ([doc](https://devcenter.heroku.com/changelog-items/630)) on Heroku? Justin, from the thread I found where you were looking into this before, it sounds like the variable is only available during the compilation phase of building the app. 

Therefore, our code to grab it would need to go in some sort of buildpack. Modifying the buildpack is probably a bad idea, but we can add our own custom buildpack that does ONLY this special thing. Heroku is apparently capable of running multiple buildpacks sequentially for the same build.

Actually, it looks like we are already running multiple buildpacks because of the need for both node and ruby. And this guy's project seems to be doing exactly what I was saying above: https://github.com/sreid/heroku-buildpack-sourceversion

I can make the necessary changes very easily if this is the route we want to go.